src/org/mozilla/javascript/Context.java

```diff
@@ -505,7 +505,7 @@ public static void exit()
      * @return The result of {@link ContextAction#run(Context)}.
      */
     @Deprecated
-    public static Object call(ContextAction action)
+    public static <T> T call(ContextAction<T> action)
```

Comment:
This is a public interface of Rhino that's probably used by a lot of projects. I don't think that this actually changes the public interface (it should just introduce a compiler warning) but what did you observe? I see that you had to change a lot of tests... I am in general trying to be very careful about the impact of changes to the public interface like this.

Reply:
This type signature just gets erased to Object - it's the same as when Java collection classes got retrofitted for types with Java 1.5, so it's all binary compatible. The benefit of this is that in most cases users of the call invocation no longer have to be casting returns, like [here](https://github.com/mozilla/rhino/pull/439/files#diff-7a7863311e0e96498e06c552114aed15L612) and [here](https://github.com/mozilla/rhino/pull/439/files#diff-ca6960f128d5cd5d051c4fba4daae2f3L37).

Recompiling against this change might cause warnings (mostly in IDEs) that some casts are no longer necessary. In general, the less explicit casts you need in your code, the better.

As for tests, I did not have to change them; they would've passed unchanged. I just lambdafied the ContextActions in them. Ability to express ContextActions as lambdas is independent of having  a type parameter on ContextAction; on the other hand, lambdas save us from having to specify `ContextAction<Object>` instead of `ContextAction` to avoid raw type warnings. So I mostly lambdafied them to avoid having to have `ContextAction<Object>` everywhere.